das/daser.go

```diff
@@ -75,6 +117,110 @@ func (d *DASer) sampling(ctx context.Context, sub header.Subscription) {
 			continue
 		}
 
+		// If the next header coming through gossipsub is not adjacent
+		// to our last DASed header, kick off routine to DAS all headers
+		// between last DASed header and h. This situation could occur
+		// either on start or due to network latency/disconnection.
+		if h.Height > checkpoint+1 {
+			// DAS headers between last DASed height up to the current
+			// header
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				d.jobsCh <- &catchUpJob{
+					from: checkpoint,
+					to:   h.Height - 1,
+				}
+			}
```

Comment:
:nooooo:
The write should be a select case, otherwise, the problem I mentioned still might be the case.

Reply:
resolved via https://github.com/celestiaorg/celestia-node/pull/473/commits/1ed26f5d47b9495e0f0d650148cae5fb0378b7d7 ?